src/pages/patientView/PatientViewPageUtils.tsx

```diff
@@ -51,3 +64,57 @@ export function getSamplesProfiledStatus(
         notProfiledIds,
     };
 }
+export function retrieveMutationalSignatureMap(
+    inputMetaData: GenericAssayMeta[]
+): MutationalSignatureLabelMap[] {
+    const mappedData = inputMetaData.map((metaData: GenericAssayMeta) => {
+        const nameSig: string =
+            'MUTATION_TYPE' in metaData.genericEntityMetaProperties
+                ? metaData.genericEntityMetaProperties['MUTATION_TYPE']
+                : '';
+        const classSig: string =
+            'MUTATION_CLASS' in metaData.genericEntityMetaProperties
+                ? metaData.genericEntityMetaProperties['MUTATION_CLASS']
+                : '';
+        const mutNameSig: string =
+            'NAME' in metaData.genericEntityMetaProperties
+                ? metaData.genericEntityMetaProperties['NAME']
+                : '';
+        const signatureId = metaData.stableId;
+        return {
+            stableId: signatureId,
+            signatureLabel: nameSig,
+            signatureClass: classSig,
+            name: mutNameSig,
+        };
+    });
+    return mappedData;
+}
+
+export function createMutationalCountsObject(
+    inputData: any,
+    signatureLabelMap: MutationalSignatureLabelMap[]
+) {
+    let result = [];
+    if (inputData && inputData.length > 0) {
+        for (const count of inputData) {
+            let mutationalSignatureChartData: IMutationalCounts = {} as IMutationalCounts;
+            mutationalSignatureChartData.patientId = count.patientId;
+            mutationalSignatureChartData.sampleId = count.sampleId;
+            mutationalSignatureChartData.studyId = count.studyId;
+            mutationalSignatureChartData.uniquePatientKey =
+                count.uniquePatientKey;
+            mutationalSignatureChartData.uniqueSampleKey =
+                count.uniqueSampleKey;
+            mutationalSignatureChartData.version =
+                _.last(count.molecularProfileId.split('_')) || '';
```

Comment:
Could we extract logic here to make it a util function, e.g. `getMutationalSignatureVersionFromProfileId`? I believe this function will be shared across lots of files.

Reply:
Extracted this logical to the proposed function `getMutationalSignatureVersionFromProfileId`
```
export function getMutationalSignaturesVersionFromProfileId(inputProfileId:string):string{
    return _.last(inputProfileId.split('_')) || '';
}
```